relay-server/src/services/store.rs

```diff
@@ -989,18 +992,104 @@ impl StoreService {
+                    seconds: safe_timestamp(received_at) as i64,
+                    nanos: 0,
+                }),
+                retention_days: retention_days.into(),
+                timestamp: Some(Timestamp {
+                    seconds: timestamp_seconds as i64,
+                    nanos: timestamp_nanos as i32,
+                }),
+                trace_id: log.trace_id.to_string(),
+                item_id,
+                attributes: Default::default(),
+                client_sample_rate: 1.0,
+                server_sample_rate: 1.0,
+            };
+
+            trace_item.attributes.insert(
+                "sentry.timestamp_precise".to_string(),
+                AnyValue {
+                    value: Some(Value::IntValue(log.timestamp_nanos as i64)),
+                },
+            );
+            trace_item.attributes.insert(
+                "sentry.severity_text".to_string(),
+                AnyValue {
+                    value: Some(Value::StringValue(
+                        log.severity_text.unwrap_or_else(|| "INFO".into()).into(),
+                    )),
+                },
+            );
+            trace_item.attributes.insert(
+                "sentry.severity_number".to_string(),
+                AnyValue {
+                    value: Some(Value::IntValue(
+                        log.severity_number.unwrap_or_default().into(),
+                    )),
+                },
+            );
+            trace_item.attributes.insert(
+                "sentry.body".to_string(),
+                AnyValue {
+                    value: Some(Value::StringValue(log.body.to_string())),
+                },
+            );
+
+            for (name, attribute) in log.attributes.unwrap_or_default() {
+                if let Some(attribute_value) = attribute {
+                    if let Some(v) = attribute_value.value {
+                        let any_value = match v {
+                            LogAttributeValue::String(value) => AnyValue {
+                                value: Some(Value::StringValue(value)),
+                            },
+                            LogAttributeValue::Int(value) => AnyValue {
+                                value: Some(Value::IntValue(value)),
+                            },
+                            LogAttributeValue::Bool(value) => AnyValue {
+                                value: Some(Value::BoolValue(value)),
+                            },
+                            LogAttributeValue::Double(value) => AnyValue {
+                                value: Some(Value::DoubleValue(value)),
+                            },
+                            LogAttributeValue::Unknown(_) => continue,
+                        };
+
+                        trace_item.attributes.insert(name.into(), any_value);
+                    }
+                }
+            }
+
+            let mut message = vec![];
+
+            if trace_item.encode(&mut message).is_err() {
```

Comment:
I think this should happen in `KafkaMessage::serialize`, store the proto type in `KafkaMessage`, and serialize in the method.

Reply:
Should be able to `#[serde(skip)]` it, then you don't have that problem: https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=0c097ebff665c8ca0bdf591c7981005f